src/main/java/de/tum/in/www1/artemis/web/rest/ResultResource.java

```diff
@@ -232,6 +232,17 @@ public ResultResource(UserService userService,
         List<Result> results = new ArrayList<>();
         Participation participation = participationService.findOne(participationId);
 
+        // Manually link the programming Exercise
```

Comment:
I do not really understand why we need these changes?
What happens if this is a "normal" participation that is not linked a programming exercise via template or solution?

Reply:
It's no longer needed as the linking now happens when creating the exercise.